diff --git a/ndpi-netfilter/ipt/libxt_ndpi.c b/ndpi-netfilter/ipt/libxt_ndpi.c
index a8fa7b7..44db092 100644
--- a/ndpi-netfilter/ipt/libxt_ndpi.c
+++ b/ndpi-netfilter/ipt/libxt_ndpi.c
@@ -382,9 +382,21 @@ static int cmp_pname(const void *p1, const void *p2) {
 	return 0;
 }
 
+static char *strupr(char *s)
+{
+	char *tmp = s;
+
+	for (; *tmp; ++tmp) {
+		*tmp = toupper((unsigned char) *tmp);
+	}
+
+	return s;
+}
+
 static int ndpi_print_prot_list(int cond) {
         int i,c,d,l;
 	char line[128];
+	char name[32];
 	char *pn[NDPI_NUM_BITS+1];
 
 	bzero((char *)&pn[0],sizeof(pn));
@@ -398,6 +410,13 @@ static int ndpi_print_prot_list(int cond) {
 	    pn[i-1] = prot_short_str[i];
 	}
 	qsort(&pn[0],NDPI_NUM_BITS,sizeof(pn[0]),cmp_pname);
+        for (i = 0; i <= NDPI_NUM_BITS; i++) {
+	    if(!pn[i]) break;
+	    strcpy(name, pn[i]);
+	    printf("--%s Match for %s protocol packets.\n", pn[i], strupr(name));
+	}
+#if 0
+
         for (i = 0,c = 0,l=0; i < NDPI_NUM_BITS; i++) {
 	    if(!pn[i]) break;
 	    l += snprintf(&line[l],sizeof(line)-1-l,"%-20s ", pn[i]);
@@ -408,6 +427,7 @@ static int ndpi_print_prot_list(int cond) {
 	    }
 	}
 	if(c > 0) printf("%s\n",line);
+#endif
 	return d;
 }
 
