--- a/ndpi-netfilter/src/main.c
+++ b/ndpi-netfilter/src/main.c
@@ -258,7 +258,7 @@ static inline int flow_have_info( struct
 static ndpi_protocol_nf proto_null = NDPI_PROTOCOL_NULL;
 
 unsigned long int ndpi_flow_limit=10000000; // 4.3Gb
-unsigned long int ndpi_enable_flow=0;
+unsigned long int ndpi_enable_flow=1;
 unsigned long int ndpi_log_debug=0;
 #ifdef NDPI_ENABLE_DEBUG_MESSAGES
 static unsigned long  ndpi_lib_trace=0;
