--- a/src/lib/third_party/src/hll/MurmurHash3.c
+++ b/src/lib/third_party/src/hll/MurmurHash3.c
@@ -44,15 +44,18 @@ u_int32_t MurmurHash(const void *key, u_
     case 3:
       k1 ^= (u_int32_t)tail[2] << 16;
       /* fall-through */
+      break;
     case 2:
       k1 ^= (u_int32_t)tail[1] << 8;
       /* fall-through */
+      break;
     case 1:
       k1 ^= tail[0];
       k1 *= c1;
       k1 = ROTL32(k1, 15);
       k1 *= c2;
       h1 ^= k1;
+      break;
     };
 
   h1 ^= len;
