Index: ndpi-netfilter-21082025-flow_info-5/ndpi-netfilter/src/main.c
===================================================================
--- ndpi-netfilter-21082025-flow_info-5.orig/ndpi-netfilter/src/main.c
+++ ndpi-netfilter-21082025-flow_info-5/ndpi-netfilter/src/main.c
@@ -80,6 +80,16 @@ extern ndpi_protocol_match host_match[];
 /* Only for debug! */
 //#define NDPI_IPPORT_DEBUG
 
+/* Workaround for MIPS: redefine xchg to simple assignment */
+#ifdef CONFIG_MIPS
+#undef xchg
+#define xchg(ptr, x) ({\
+    typeof(*(ptr)) __old = *(ptr);\
+    *(ptr) = (x);\
+    __old;\
+})
+#endif
+
 
 #define COUNTER(a) atomic_inc((atomic_t *)&a)
 #define COUNTER_D(a) atomic_dec((atomic_t *)&a)
