--- a/Makefile.Linux
+++ b/Makefile.Linux
@@ -8,13 +8,12 @@
 # library support. Add -DSAFESQL for poorely written ODBC library / drivers.
 
 BUILDDIR = ../bin/
-CC = gcc
 
-CFLAGS = -g  -fPIC -O2 -fno-strict-aliasing -c -pthread -DWITHSPLICE -D_GNU_SOURCE -DGETHOSTBYNAME_R -D_THREAD_SAFE -D_REENTRANT -DNOODBC -DWITH_STD_MALLOC -DFD_SETSIZE=4096 -DWITH_POLL -DWITH_NETFILTER
+CFLAGS = -fPIC -fno-strict-aliasing -c -pthread -DWITHSPLICE -D_GNU_SOURCE -DGETHOSTBYNAME_R -D_THREAD_SAFE -D_REENTRANT -DNOODBC -DWITH_STD_MALLOC -DFD_SETSIZE=4096 -DWITH_POLL -DWITH_NETFILTER -DSYSTEM_PCRE
 COUT = -o 
 LN = $(CC)
 DCFLAGS = 
-LDFLAGS = -fPIC -O2 -fno-strict-aliasing -pthread
+LDFLAGS = -fno-strict-aliasing 
 DLFLAGS = -shared
 DLSUFFICS = .ld.so
 # -lpthreads may be reuqired on some platforms instead of -pthreads
@@ -35,7 +34,7 @@ MAKEFILE = Makefile.Linux
 #LIBS = -lcrypto -lssl -ldl 
 LIBS = -ldl 
 #PLUGINS = SSLPlugin StringsPlugin TrafficPlugin PCREPlugin TransparentPlugin PamAuth
-PLUGINS = StringsPlugin TrafficPlugin PCREPlugin TransparentPlugin
+PLUGINS = SSLPlugin StringsPlugin TrafficPlugin PCREPlugin TransparentPlugin
 
 include Makefile.inc
 
Index: 3proxy/src/sockmap.c
===================================================================
--- 3proxy.orig/src/sockmap.c
+++ 3proxy/src/sockmap.c
@@ -19,10 +19,11 @@ char logbuf[1024];
 #define log(X)
 #endif
 
+#define _FILE_OFFSET_BITS 64
 #ifdef WITHSPLICE
 
 #include <fcntl.h>
-ssize_t splice(int fd_in, loff_t *off_in, int fd_out, loff_t *off_out, size_t len, unsigned int flags);
+ssize_t splice(int fd_in, off_t *off_in, int fd_out, off_t *off_out, size_t len, unsigned int flags);
 #ifndef SPLICE_F_MOVE
 #define SPLICE_F_MOVE           0x01
 #endif
