Index: mtproxy/mtproto/mtproto-proxy.c
===================================================================
--- mtproxy.orig/mtproto/mtproto-proxy.c
+++ mtproxy/mtproto/mtproto-proxy.c
@@ -68,7 +68,7 @@
 #define COMMIT "unknown"
 #endif
 
-#define VERSION_STR	"mtproxy-3.0.5"
+#define VERSION_STR	"mtproxy-3.0.6"
 const char FullVersionStr[] = VERSION_STR " compiled at " __DATE__ " " __TIME__ " by gcc " __VERSION__ " "
 #ifdef __LP64__
   "64-bit"
